### PR TITLE
[skip ci] Added a README.md file in requirements folder 

### PR DESCRIPTION
**Pull request by csking101**
# Pull Request Template

## Description

Fix #3224

Fixed a typo in the CONTRIBUTING.md file. In line 36, "seperate" has been corrected to "separate".
Added a README.md file in the requirements folder as per #2819.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by gusty1g** [comment]
I consider this PR to be too trivial to consider as a contribution. There are some documentation [issues](https://github.com/deepchem/deepchem/issues?q=is%3Aopen+is%3Aissue+label%3Adocs) open. Could you consider adding them together with these changes?

**Comment by csking101** [comment]
Sure, I will look into the other issues and try to club it with these changes.

**Comment by csking101** [comment]
@gusty1g Could you have a look at the newly created README.md file in the requirements folder.

**Comment by csking101** [comment]
@gusty1g What can I do to fix these failing tests?

**Comment by rbharath** [comment]
@csking101 Our review etiquette is to not repeatedly tag maintainers before at least a week has passed without a response. 

The new `README.md` file looks useful at first look. If you can stop by our OH at some point, I am glad to help with a more thorough review

**Comment by csking101** [comment]
> @csking101 Our review etiquette is to not repeatedly tag maintainers before at least a week has passed without a response. 
> 
> The new `README.md` file looks useful at first look. If you can stop by our OH at some point, I am glad to help with a more thorough review

I am extremely sorry, actually this is my first time working with an open source organization. I was just unsure about what to do with the failed test, since I had not written any code.

Thank you, would the OH be held on gitter? Also, where are the meeting dats announced?

File: requirements/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Requirements
+DeepChem is a python deep-learning library for molecular machine learning. During its installation, it requires various prerequisite libraries to be installed. The various `.yml` files in the `requirements` folder contain the different libraries that are required based on the installer. The different folders are used to show the dependencies of the DeepChem library on the respective deep-learning backends.
```
Comment (gusty1g) [inline requirements/README.md:2]: Can we remove this line because 1. we don't need what deepchem is about here (the first line) 2. it is kind of obvious that requirements folder contains requirements (third line).

File: requirements/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+The files present are:
+1. `env_common.yml` - contains libraries related to scientific applications that are common to all operating systems.
```
Comment (gusty1g) [inline requirements/README.md:5]: this is good - what different env files are

File: requirements/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+2. `env_dqc.yml` - contains libraries specific for using Quantum Chemistry packages through the [Differentiable Quantum Chemistry package](https://github.com/diffqc/dqc).
+3. `env_mac.yml` - contains dependencies specific to macOS.
+4. `env_ubuntu.yml` - contains dependencies specific to Ubuntu.
+5. `env_test.yml` - contains libraries related to testing code and models.
+
+The folders present are:
```
Comment (gusty1g) [inline requirements/README.md:11]: we don't need here what jax or pytorch or tensorflow here is. The below three points can be compressed into a single line, something like the three subfolder contains cpu and gpu deep learning backend requirements of deepchem.

File: requirements/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+2. `flaky` - [Flaky](https://github.com/box/flaky) is a plugin for nose or pytest that automatically reruns flaky tests.
+3. `mypy` - [Mypy](https://github.com/python/mypy) is a static type checker for Python.
+4. `pytest` - The [pytest framework](https://github.com/pytest-dev/pytest) makes it easy to write small tests.
+5. `pytest-cov` - It's a coverage [plugin](https://github.com/pytest-dev/pytest-cov) for `pytest`.
+6. `types-pkg_resources` - This is an auto-generated PEP 561 type stub package for pkg\_resources package. It can be used by type-checking tools like mypy, PyCharm, pytype etc. to check code that uses pkg\_resources. Its source is [here](https://github.com/python/typeshed/tree/master/stubs/pkg_resources).
+7. `types-setuptools` - This is a PEP 561 type stub package for the setuptools package. It can be used by type-checking tools like mypy, pyright, pytype, PyCharm, etc. to check code that uses setuptools. Its source is [here](https://github.com/python/typeshed/tree/main/stubs/setuptools).
+8. `yapf` - It is a [formatter](https://github.com/google/yapf) for Python files.
+
+## jax
+### env_jax.cpu.yml
```
Comment (gusty1g) [inline requirements/README.md:74]: In this part, probably we don't need such a long description of what each requirements files are about. Can you make it short, something like a one line description for each file, like `env_jax.cpu.yml` - specifies jax installation requirements for cpu.

Following the same for tensorflow and torch will be helpful.

File: deepchem/models/torch_models/kfac_optimizer.py [hunk 5]
```diff
@@ -1,486 +1,486 @@
...
+try:
+    import torch
+    import torch.optim as optim
+    has_pytorch = True
+
+except ModuleNotFoundError:
+    has_pytorch = False
+
+
+class KFACOptimizer(optim.Optimizer):
```
Comment (gusty1g) [inline deepchem/models/torch_models/kfac_optimizer.py:15]: I am not sure here why by changing a single typo, git is showing diff in many other lines, though the visible difference between other lines seems to be none.

File: requirements/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
+# Requirements
+During its installation, DeepChem requires various prerequisite libraries to be installed. The various `.yml` files in the `requirements` folder contain the different libraries that are the prerequisites based on the installer, operating system, etc. The different folders are used to show the dependencies of the DeepChem library on the respective deep-learning backends.
```
Comment (gusty1g) [inline requirements/README.md:2]: How about something like this - DeepChem depends on various scientific computational libraries which are specified in the `.yml` files. The different sub-folder contains the deep-learning backend requirements of DeepChem.

**Comment by gusty1g** [comment]
Apart from the comments, the rest of the documentation looks good to me.

File: deepchem/models/torch_models/kfac_optimizer.py (hunk 5 above)
Comment (csking101) [reply deepchem/models/torch_models/kfac_optimizer.py:15]: Yes, I am not able to find anything online for why it's happening.

**Comment by gusty1g** [comment]
Mostly it looks good to me. I still need to figure out the reason for changes in `kfac_optimizer.py` - why for a single line of change, git is showing a big diff for merging this.
